hydra-explorer/src/Hydra/Explorer/ExplorerState.hs

```diff
@@ -0,0 +1,197 @@
+module Hydra.Explorer.ExplorerState where
+
+import Hydra.Prelude
+
+-- XXX: Depending on hydra-node will be problematic to support versions
+import Hydra.HeadId (HeadId (..))
+
+import Hydra.Cardano.Api (TxIn, TxOut)
+import Hydra.Cardano.Api.Prelude (CtxUTxO)
+import Hydra.Chain.Direct.Tx (
+  AbortObservation (..),
+  CloseObservation (..),
+  CollectComObservation (..),
+  CommitObservation (..),
+  ContestObservation (..),
+  FanoutObservation (..),
+  HeadObservation (..),
+  InitObservation (..),
+ )
+import Hydra.ContestationPeriod (ContestationPeriod)
+import Hydra.OnChainId (OnChainId)
+import Hydra.Party (Party)
+
+data PartyCommit = PartyCommit
+  { txIn :: TxIn
+  , txOut :: TxOut CtxUTxO
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
+
+instance Arbitrary PartyCommit where
+  arbitrary = genericArbitrary
+
+data HeadMember = HeadMember
+  { party :: Party
+  , onChainId :: OnChainId
+  , commits :: [PartyCommit]
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
+
+instance Arbitrary HeadMember where
+  arbitrary = genericArbitrary
+
+data HeadStatus
+  = Initializing
+  | Aborted
+  | Open
+  | Closed
+  | Finalized
+  deriving stock (Eq, Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
+
+instance Arbitrary HeadStatus where
+  arbitrary = genericArbitrary
+
+data HeadState = HeadState
+  { headId :: HeadId
+  , seedTxIn :: TxIn
+  , status :: HeadStatus
+  , contestationPeriod :: ContestationPeriod
+  , members :: [HeadMember]
```

Comment:
Idea (not needed right away):

```suggestion
  , members :: [HeadMember]
  , contesters :: [Party]
```

Reply:
The json instance for Plutus.PubKeyHash gives an object and not a plain string.

Are u ok to use this type as part of the openapi; or shall I use Text and use the Show instance for Plutus.PubKeyHash?

Reply:
used `contestations :: Observed Natural` as agreed